scalalib/src/publish/SonatypePublisher.scala

```diff
@@ -154,12 +152,9 @@ class SonatypePublisher(uri: String,
   }
 
   // http://central.sonatype.org/pages/working-with-pgp-signatures.html#signing-a-file
-  private def poorMansSign(file: os.Path, maybePassphrase: Option[String], maybeKeyName: Option[String]): os.Path = {
+  private def gpgSigned(file: os.Path, args: Seq[String]): os.Path = {
     val fileName = file.toString
-    val optionFlag = (flag: String, ov: Option[String]) => ov.map(flag :: _ :: Nil).getOrElse(Nil)
-    val command = "gpg" ::
-      optionFlag("--passphrase", maybePassphrase) ++ optionFlag("-u", maybeKeyName) ++
-        Seq("--batch", "--yes", "-a", "-b", fileName)
+    val command = Seq("gpg", "--batch", "--yes", "-a", "-b", fileName) ++ args
```

Comment:
Wondering, if we should move these defaults into the default param to `gpgArgs` and document it's meaning. It's not a change request, just a thought. WDYT?

Reply:
I was thinking the same but then people need to be aware that if there specify the `--gpgArgs` they need to provide these too to keep them.

Reply:
We could add a hint in the scaladoc, too. But having these hard-coded args screams for future maintenance work, as seen in the past (#345, #530, #753). We could also rename the parameter to `alternativeGpgArgs` or `overrideGpgArgs` to indicate the fact that it replaced all defaults.

Reply:
What about an accessor to the defaults to make reusing easy:

```scala
val defaultGpgArgs = ...
def publish(...,  gpgArgs: Seq[String] = defaultGpgArgs, ...)
```

And a simple example in the docs, showing how to add a passphrase using the defaultsArgs + the passphrase args

Reply:
Sounds good to me. What do you think @jodersky?

Reply:
Looks good! I agree with @lefou's suggestion too. Thanks for working on this!

Reply:
Change pushed